PJ05/src/Exceptions/RequestParsingException.java

```diff
@@ -1,13 +1,10 @@
 package Exceptions;
 
-public class RequestParsingException extends Exception
-{
-    public RequestParsingException()
-    {
+public class RequestParsingException extends RequestFailedException {
+    public RequestParsingException() {
```

Comment:
This is **NOT** a `RequestFailedException`. 

A `RequestFailedException` should be sent back to the client to be handled, since it's the client's responsibility to send reasonable requests.

`RequestParsingException` is regarding the error of the server in parsing any request, which is NOT expected to happen anytime provided our code is correct, and would probably cause a system crash if it does happen.

If some worker receives a corrupted request, we should terminate that `ServerWorker` and kick that client.